### PR TITLE
Fixes not equals on numbers

### LINKED ISSUE #16: Adding uniqueness constraint results in non-deduped entries
Added the following validation to our `QueueItem` model:
```ruby
validates :incoming_fax_id, uniqueness: true
```
resulted in
```sql
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 1) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 11) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 12) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 13) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 14) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 15) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 16) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 17) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 18) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 19) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 22) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 38) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 39) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 4) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 42) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 43) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 44) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 5) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 50) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 51) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 53) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 54) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 6) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 66) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 67) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 68) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 69) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 7) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 70) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 71) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" = number-redacted AND "queue_items"."id" != 84) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 1) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 15) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 16) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 17) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 18) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 2) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 28) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 29) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 3) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 30) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 31) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 32) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 33) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 34) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 35) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 36) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 4) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 5) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 68) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 69) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 70) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 71) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 72) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 73) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 74) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 75) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 76) LIMIT 1
+SELECT  1 AS one FROM "queue_items" WHERE ("queue_items"."incoming_fax_id" IS NULL AND "queue_items"."id" != 83) LIMIT 1
```

Old and new footprints: [footprints.zip](https://github.com/covermymeds/sql_footprint/files/215053/footprints.zip)

### PR DESCRIPTION
**Pull request by brandonjoyce**
Resolves #16

### CODE REVIEW

**Comment by schneiderderek** [comment]
LGTM :+1: 

Though we might want to consider an elegant way to handle other operators since we know there are others.

File: lib/sql_footprint/sql_anonymizer.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@ class SqlAnonymizer
     GSUBS = {
       /\sIN\s\((.*)\)/ => ' IN (values-redacted)'.freeze, # IN clauses
       /\s\=\s([0-9]+)/ => ' = number-redacted'.freeze, # numbers
+      /\s\!\=\s([0-9]+)/ => ' != number-redacted'.freeze, # numbers w/ not equals
```
Comment (ryanzbone) [inline lib/sql_footprint/sql_anonymizer.rb]: would something like `/\s.\=\s([0-9]+)/` give us what @schneiderderek is talking about? 
or `/\s[!<>]{0,1}\=\s([0-9]+)/` 
Then maybe you could capture the thing to put it in the number-redacted string?

Comment (schneiderderek) [inline lib/sql_footprint/sql_anonymizer.rb]: @ryanzbone I like the idea. I think you still have a bug or two in your regex. It wouldn't capture `<`, or `>`. But, I think this regex might work better: `/\s+(!=|=|<|>|<=|>=)\s+[0-9]+/. It's a little more explicit.

Comment (schneiderderek) [inline lib/sql_footprint/sql_anonymizer.rb]: We would also need to rewrite some of the code for the sql anonymizer, but I don't think that would be difficult.

**Comment by brandonjoyce** [comment]
Well here's the inelegant fix w/ more operators.  You ok with this for now? @schneiderderek @ryanzbone

File: lib/sql_footprint/sql_anonymizer.rb [hunk 2]
```diff
@@ -1,11 +1,14 @@
 module SqlFootprint
   class SqlAnonymizer
+    gsubs = {}
+    ['>', '<', '!=', '<=', '>='].each do |operator|
+      gsubs[/\s#{Regexp.quote(operator)}\s([0-9]+)/] = " #{operator} number-redacted"
```
Comment (mikegee) [inline lib/sql_footprint/sql_anonymizer.rb]: @schneiderderek's regex using a capture ground simplifies this quite a bit:

```ruby
/\s+(!=|=|<|>|<=|>=)\s+[0-9]+/ => ' \1 number-redacted'
```

**Comment by schneiderderek** [comment]
@brandonjoyce I prefer the regex that I used rather than iterating and making a bunch of additional regexs.

**Comment by brandonjoyce** [comment]
Changed to derek's regex method.  Good to go? @schneiderderek

**Comment by mikegee** [comment]
LGTM

**Comment by schneiderderek** [comment]
:+1:

**Pull request merged**